grails-app/services/org/pih/warehouse/product/ProductClassificationService.groovy

```diff
@@ -0,0 +1,55 @@
+package org.pih.warehouse.product
+
+import grails.gorm.transactions.Transactional
+
+import org.pih.warehouse.inventory.InventoryLevel
+
+/**
+ * Product Classification is a way of categorizing products for the purposes of measuring their impact/importance,
+ * which in turn can be used for prioritization/configuration of feature-specific tasks.
+ *
+ * For example, the stock of fast moving items is more volatile compared to other items. They also often account for
+ * a higher percentage of the financial value of the products that are moving through a facility. As such, we may want
+ * to view them as more "important", which we designate via a classification ("A" for example). Rules can then be
+ * applied to the classification as needed. For instance, we can decide that all "A" products should be be counted at
+ * least once a week.
+ *
+ * It's important to note that the classifications themselves contain no inherent value. It's up to the implementers
+ * to decide what a classification really means, which is determined by the feature-specific rules that are configured
+ * for each classification.
+ *
+ * For example, if you're using the ABC Classification rules, the classes "A", "B", "C" are just strings.
+ * The meaning comes in when we configure the cycle count feature to require class "A" products to be counted weekly,
+ * class "B" products to be counted monthly, and class "C" products to be counted yearly.
+ */
+@Transactional
```

Comment:
I don't remember our convention, but we might want to make the list() method @Transactional(readOnly=true)?